src/main/java/com/szeloch/jan/parkingsystemmanagement/rest/controller/parkingclient/ParkingClientsController.java

```diff
@@ -0,0 +1,192 @@
+                .findAll()
+                .stream()
+                .map(ParkingClientResource::new)
+                .collect(Collectors.toList());
+
+        return new Resources<>(parkingClients,
+                linkTo(methodOn(ParkingClientsController.class).findAll()).withSelfRel());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ParkingClientResource> findOne(@PathVariable final long id) {
+
+        return clientsRepository
+                .findById(id)
+                .map(parkingClient -> ResponseEntity.ok(new ParkingClientResource(parkingClient)))
+                .orElseThrow(() -> new ParkingClientNotFoundException(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<ParkingClientResource> newParkingClient(@RequestBody final ParkingClient newParkingClient) {
+
+        ParkingClientResource resource = new ParkingClientResource(clientsRepository.save(newParkingClient));
+
+        return ResponseEntity
+                .created(URI.create(resource.getId().expand().getHref()))
+                .body(resource);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<ParkingClientResource> updateOrReplaceParkingClient(@RequestBody ParkingClient newParkingClient, @PathVariable final Long id) {
+
+        ParkingClient updatedParkingClient = clientsRepository.findById(id)
+                .map(parkingClient -> {
+                    parkingClient.setFirstName(newParkingClient.getFirstName());
+                    parkingClient.setLastName(newParkingClient.getLastName());
+                    parkingClient.setAddress(newParkingClient.getAddress());
+                    return clientsRepository.save(parkingClient);
+                })
+                .orElseGet(() -> {
+                    newParkingClient.setId(id);
+                    return clientsRepository.save(newParkingClient);
+                });
+
+        ParkingClientResource resource = new ParkingClientResource(updatedParkingClient);
+
+        return ResponseEntity
+                .created(URI.create(resource.getId().expand().getHref()))
+                .body(resource);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteParkingClient(@PathVariable final Long id) {
+
+        return clientsRepository
+                .findById(id)
+                .map(parkingClient -> {
+                    clientsRepository.deleteById(id);
+                    return ResponseEntity.noContent().build();
+                })
+                .orElseThrow(() -> new ParkingClientNotFoundException(id));
+    }
+
+    @GetMapping("/{id}/vehicles")
+    public Resources<Resource<Vehicle>> findAllParkingClientVehicles(@PathVariable final Long id) {
+
+        List<Resource<Vehicle>> parkingClientsVehicles = vehiclesRepository.findByOwnerId(id).stream()
+                .map(vehicle -> new Resource<>(vehicle,
+                        ControllerLinkBuilder.linkTo(methodOn(VehiclesController.class).findOne(vehicle.getId())).withSelfRel(),
```

Comment:
Mało czytelna ta ściana kodu.

Reply:
Rzeczywiście, można by to było trochę rozbić dla poprawy czytelności.